src/control/client/mocks.go

```diff
@@ -358,56 +361,70 @@ func (m *mockControl) logger() logging.Logger {
 
 func newMockControl(
 	log logging.Logger,
-	address string, state connectivity.State, connectRet error,
+	address string, cfg mockControlConfig,
 	cClient ctlpb.MgmtCtlClient, sClient mgmtpb.MgmtSvcClient) Control {
 
-	return &mockControl{address, state, connectRet, cClient, sClient, log}
+	return &mockControl{address, cfg, cClient, sClient, log}
 }
 
 type mockControllerFactory struct {
-	log             logging.Logger
-	state           connectivity.State
-	ctrlrs          NvmeControllers
-	ctrlrResults    NvmeControllerResults
-	modules         ScmModules
-	moduleResults   ScmModuleResults
-	pmems           ScmNamespaces
-	mountResults    ScmMountResults
-	scanRet         error
-	formatRet       error
-	killRet         error
-	connectRet      error
-	getACLResult    *mockACLResult
-	listPoolsResult *mockListPoolsResult
+	log          logging.Logger
+	controlCfg   mockControlConfig
+	ctlClientCfg mockMgmtCtlClientConfig
+	svcClientCfg mockMgmtSvcClientConfig
 }
 
 func (m *mockControllerFactory) create(address string, cfg *security.TransportConfig) (Control, error) {
 	// returns controller with mock properties specified in constructor
-	cClient := newMockMgmtCtlClient(m.ctrlrs, m.ctrlrResults, m.modules,
-		m.moduleResults, m.pmems, m.mountResults, m.scanRet, m.formatRet)
+	cClient := &mockMgmtCtlClient{cfg: m.ctlClientCfg}
 
-	sClient := newMockMgmtSvcClient(m.getACLResult, m.listPoolsResult)
+	sClient := &mockMgmtSvcClient{cfg: m.svcClientCfg}
 
-	controller := newMockControl(m.log, address, m.state, m.connectRet, cClient, sClient)
+	controller := newMockControl(m.log, address, m.controlCfg, cClient, sClient)
 
 	err := controller.connect(address, cfg)
 
 	return controller, err
 }
 
+// TODO: switch everything over to using this
```

Comment:
I like this. Felt like the mock factory methods were getting out of hand, thanks for addressing it now.